[live555] build failure

**Host Environment**
- OS: [Windows]
- Compiler: MSVC 19.16.27045

**To Reproduce**
`./vcpkg install live555 --head`

**Failure logs** 
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/6282509/install-x64-windows-dbg-out.log)

**Additional context**
none

```
F:\dev-tool\vcpkg-report\buildtrees\live555\src\555-latest-775e7a0b73.clean\groupsock\GroupsockHelper.cpp(46): fatal error C1083: Cannot open include file: 'ifaddrs.h': No such file or directory
```

I patched the live555 CMakeLists.txt as described, but then I get this other compile error:

C:\Project\_tools\vcpkg\buildtrees\live555\src\555-latest-9b8fcaac25.clean\groupsock\GroupsockHelper.cpp(183): error C2664: 'int setsockopt(SOCKET,int,int,const char *,int)': cannot convert argument 4 from 'const int *' to 'const char *'

--- a/ports/live555/CMakeLists.txt
+++ b/ports/live555/CMakeLists.txt
@@ -8,6 +8,10 @@ include_directories(
     UsageEnvironment/include
 )
 
+if (WIN32)
+    add_definitions(-DNO_GETIFADDRS)
+endif(WIN32)
+
 if (NOT MSVC)
     add_compile_options(-DSOCKLEN_T=socklen_t)
 endif()

--- a/ports/live555/portfile.cmake
+++ b/ports/live555/portfile.cmake
@@ -1,16 +1,9 @@
 vcpkg_check_linkage(ONLY_STATIC_LIBRARY) 
 
-if(NOT VCPKG_USE_HEAD_VERSION)
-    # Live555 only makes the latest releases available for download on their site
-    message(FATAL_ERROR "Live555 does not have persistent releases. Please re-run the installation with --head.")
-endif()
-
-set(LIVE_VERSION latest)
-
 vcpkg_download_distfile(ARCHIVE
-    URLS "http://www.live555.com/liveMedia/public/live555-${LIVE_VERSION}.tar.gz"
-    FILENAME "live555-${LIVE_VERSION}.tar.gz"
-    SKIP_SHA512
+    URLS "http://www.live555.com/liveMedia/public/live.2021.12.18.tar.gz"
+    FILENAME "live.2021.12.18.tar.gz"
+    SHA512 072ae92a1a63687e33c659627fbf562e6f2cdf73cf1c3199959aa3a713636ccd4113d56270c4793c86b42338000f180e16a7b2cf3d893402226a04360f8b2e97
 )
 
 vcpkg_extract_source_archive_ex(
@@ -35,7 +28,7 @@ file(GLOB HEADERS
     "${SOURCE_PATH}/UsageEnvironment/include/*.h*"
 )
 
-file(COPY ${HEADERS} DESTINATION ${CURRENT_PACKAGES_DIR}/include)
-file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(COPY ${HEADERS} DESTINATION "${CURRENT_PACKAGES_DIR}/include")
+file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
 
 vcpkg_copy_pdbs()

--- a/ports/live555/portfile.cmake
+++ b/ports/live555/portfile.cmake
@@ -20,14 +20,13 @@ vcpkg_extract_source_archive_ex(
         fix-RTSPClient.patch
 )
 
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
 )
 
-vcpkg_install_cmake()
+vcpkg_cmake_install()
 
 file(GLOB HEADERS
     "${SOURCE_PATH}/BasicUsageEnvironment/include/*.h*"

--- a/ports/live555/vcpkg.json
+++ b/ports/live555/vcpkg.json
@@ -1,7 +1,6 @@
 {
   "name": "live555",
-  "version-string": "latest",
-  "port-version": 3,
+  "version-date": "2021-12-18",
   "description": "A complete RTSP server application",
   "homepage": "https://www.live555.com/liveMedia",
   "dependencies": [

--- a/ports/live555/vcpkg.json
+++ b/ports/live555/vcpkg.json
@@ -1,10 +1,18 @@
 {
   "name": "live555",
   "version-string": "latest",
-  "port-version": 2,
+  "port-version": 3,
   "description": "A complete RTSP server application",
   "homepage": "https://www.live555.com/liveMedia",
   "dependencies": [
-    "openssl"
+    "openssl",
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
   ]
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4138,7 +4138,7 @@
     },
     "live555": {
       "baseline": "latest",
-      "port-version": 2
+      "port-version": 3
     },
     "llfio": {
       "baseline": "2.0-20211208",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4137,8 +4137,8 @@
       "port-version": 2
     },
     "live555": {
-      "baseline": "latest",
-      "port-version": 3
+      "baseline": "2021-12-18",
+      "port-version": 0
     },
     "llfio": {
       "baseline": "2.0-20211208",

--- a/versions/l-/live555.json
+++ b/versions/l-/live555.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d2d9c96e8a0fa3a6af648952b384dc53504bf738",
+      "version-date": "2021-12-18",
+      "port-version": 0
+    },
+    {
       "git-tree": "7d349d264f7be8a8802b521fddaa5ad047b48720",
       "version-string": "latest",
       "port-version": 2

--- a/versions/l-/live555.json
+++ b/versions/l-/live555.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "aabb9aa641359a69dba4f8df977c85c3b10c2af4",
+      "version-string": "latest",
+      "port-version": 3
+    },
+    {
       "git-tree": "7d349d264f7be8a8802b521fddaa5ad047b48720",
       "version-string": "latest",
       "port-version": 2

--- a/versions/l-/live555.json
+++ b/versions/l-/live555.json
@@ -1,11 +1,6 @@
 {
   "versions": [
     {
-      "git-tree": "aabb9aa641359a69dba4f8df977c85c3b10c2af4",
-      "version-string": "latest",
-      "port-version": 3
-    },
-    {
       "git-tree": "7d349d264f7be8a8802b521fddaa5ad047b48720",
       "version-string": "latest",
       "port-version": 2

